instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/LoggingEventToReplay.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.logback.appender.v1_0;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.LoggerContextVO;
+import io.opentelemetry.instrumentation.logback.appender.v1_0.internal.LoggingEventWithThreadId;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Marker;
+import org.slf4j.event.KeyValuePair;
+
+class LoggingEventToReplay implements LoggingEventWithThreadId {
+
+  private final ILoggingEvent loggingEvent;
+  private final long timeStamp;
+  private StackTraceElement[] callerData;
+  private String threadName;
+  private long threadId;
+
+  LoggingEventToReplay(
+      ILoggingEvent loggingEvent,
+      boolean captureExperimentalAttributes,
+      boolean captureCodeAttributes) {
+    this.loggingEvent = loggingEvent;
+    this.timeStamp = loggingEvent.getTimeStamp();
```

Comment:
does this need to be cached (or can it be calculated below since we are keeping full reference to `loggingEvent`)?

Reply:
Sure for callerData. Not sure for timestamp because not sure about the accuracy of the test for this.